target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -59,7 +59,11 @@ ipq40xx_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0u@eth0" "1:lan" "2:lan" "3:lan" "4:lan"
 		;;
-	avm,fritzrepeater-3000|\
+	avm,fritzrepeater-3000)
+                ucidef_set_interface_lan "eth0 eth1"
+                ucidef_add_switch "switch0" \
+                        "0u@eth0" "4:lan" "5:lan"
```

Comment:
This isn't correct, as port 5 is "connected" to virtual gmac1.

On the other hand, the gmac1 might be ditched altogether by updating the port mask for gmac0 and the ess-switch. This would also avoid traffic to go thru the CPU.

Reply:
Port 5 is the physical "LAN2" port, see output below. 

port 4 -> LAN1
port 5 -> LAN2

LAN-cable plugged into LAN2
```
root@avm3000:~# swconfig dev switch0 show
Global attributes:
	enable_vlan: 1
	enable_mirror_rx: 0
	enable_mirror_tx: 0
	mirror_monitor_port: 0
	mirror_source_port: 0
	linkdown: ???
Port 0:
	mib: Port 0 MIB counters
RxBroad     : 382
RxPause     : 0
RxMulti     : 43256
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 3177449
Rx256Byte   : 131067
Rx512Byte   : 52269
Rx1024Byte  : 35859
Rx1518Byte  : 33087
RxMaxByte   : 131441
RxTooLong   : 0
RxGoodByte  : 567139776
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 35445
TxPause     : 0
TxMulti     : 424
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 423423
Tx256Byte   : 109539
Tx512Byte   : 45449
Tx1024Byte  : 1925249
Tx1518Byte  : 485658
TxMaxByte   : 5412948
TxOverSize  : 0
TxByte      : 10212553931
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:0 link:up speed:1000baseT full-duplex txflow rxflow 
Port 1:
	mib: Port 1 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:1 link:down
Port 2:
	mib: Port 2 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:2 link:down
Port 3:
	mib: Port 3 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:3 link:down
Port 4:
	mib: Port 4 MIB counters
RxBroad     : 35436
RxPause     : 0
RxMulti     : 381
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 429555
Rx256Byte   : 104179
Rx512Byte   : 45178
Rx1024Byte  : 1925065
Rx1518Byte  : 545564
RxMaxByte   : 5352663
RxTooLong   : 0
RxGoodByte  : 10178934657
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 380
TxPause     : 1620
TxMulti     : 43244
TxUnderRun  : 0
Tx64Byte    : 197481
Tx128Byte   : 2996787
Tx256Byte   : 116853
Tx512Byte   : 51673
Tx1024Byte  : 35891
Tx1518Byte  : 32652
TxMaxByte   : 131416
TxOverSize  : 0
TxByte      : 546299961
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 1
	link: port:4 link:up speed:1000baseT full-duplex txflow rxflow auto
Port 5:
	mib: Port 5 MIB counters
RxBroad     : 9
RxPause     : 0
RxMulti     : 43
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 9
Rx128Byte   : 25
Rx256Byte   : 20
Rx512Byte   : 6
Rx1024Byte  : 2
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 10210
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 2
TxPause     : 0
TxMulti     : 12
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 12
Tx256Byte   : 25
Tx512Byte   : 2
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 5612
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 2
	link: port:5 link:up speed:1000baseT full-duplex txflow rxflow auto
VLAN 1:
	vid: 1
	ports: 0t 4 
VLAN 2:
	vid: 2
	ports: 0t 5 
VLAN 20:
	vid: 20
	ports: 0t 4t 
VLAN 21:
	vid: 21
	ports: 0t 4t 
VLAN 22:
	vid: 22
	ports: 0t 4t 
VLAN 23:
	vid: 23
	ports: 0t 4t 
```
LAN-cable not plugged into LAN2
```
root@avm3000:~# swconfig dev switch0 show
Global attributes:
	enable_vlan: 1
	enable_mirror_rx: 0
	enable_mirror_tx: 0
	mirror_monitor_port: 0
	mirror_source_port: 0
	linkdown: ???
Port 0:
	mib: Port 0 MIB counters
RxBroad     : 382
RxPause     : 0
RxMulti     : 43353
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 3178234
Rx256Byte   : 131229
Rx512Byte   : 52340
Rx1024Byte  : 35899
Rx1518Byte  : 33125
RxMaxByte   : 131470
RxTooLong   : 0
RxGoodByte  : 567378971
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 35491
TxPause     : 0
TxMulti     : 509
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 423984
Tx256Byte   : 109622
Tx512Byte   : 45530
Tx1024Byte  : 1925312
Tx1518Byte  : 485737
TxMaxByte   : 5413182
TxOverSize  : 0
TxByte      : 10213152499
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:0 link:up speed:1000baseT full-duplex txflow rxflow 
Port 1:
	mib: Port 1 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:1 link:down
Port 2:
	mib: Port 2 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:2 link:down
Port 3:
	mib: Port 3 MIB counters
RxBroad     : 0
RxPause     : 0
RxMulti     : 0
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 0
Rx256Byte   : 0
Rx512Byte   : 0
Rx1024Byte  : 0
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 0
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 0
TxPause     : 0
TxMulti     : 0
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 0
Tx256Byte   : 0
Tx512Byte   : 0
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 0
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 0
	link: port:3 link:down
Port 4:
	mib: Port 4 MIB counters
RxBroad     : 35462
RxPause     : 0
RxMulti     : 383
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 0
Rx128Byte   : 430062
Rx256Byte   : 104233
Rx512Byte   : 45232
Rx1024Byte  : 1925122
Rx1518Byte  : 545643
RxMaxByte   : 5352897
RxTooLong   : 0
RxGoodByte  : 10179506745
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 380
TxPause     : 1620
TxMulti     : 43341
TxUnderRun  : 0
Tx64Byte    : 197810
Tx128Byte   : 2997255
Tx256Byte   : 116984
Tx512Byte   : 51742
Tx1024Byte  : 35930
Tx1518Byte  : 32690
TxMaxByte   : 131445
TxOverSize  : 0
TxByte      : 546530624
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 1
	link: port:4 link:up speed:1000baseT full-duplex txflow rxflow auto
Port 5:
	mib: Port 5 MIB counters
RxBroad     : 29
RxPause     : 0
RxMulti     : 126
RxFcsErr    : 0
RxAlignErr  : 0
RxRunt      : 0
RxFragment  : 0
Rx64Byte    : 31
Rx128Byte   : 57
Rx256Byte   : 50
Rx512Byte   : 32
Rx1024Byte  : 8
Rx1518Byte  : 0
RxMaxByte   : 0
RxTooLong   : 0
RxGoodByte  : 32286
RxBadByte   : 0
RxOverFlow  : 0
Filtered    : 0
TxBroad     : 2
TxPause     : 0
TxMulti     : 12
TxUnderRun  : 0
Tx64Byte    : 0
Tx128Byte   : 25
Tx256Byte   : 34
Tx512Byte   : 2
Tx1024Byte  : 0
Tx1518Byte  : 0
TxMaxByte   : 0
TxOverSize  : 0
TxByte      : 8110
TxCollision : 0
TxAbortCol  : 0
TxMultiCol  : 0
TxSingleCol : 0
TxExcDefer  : 0
TxDefer     : 0
TxLateCol   : 0

	pvid: 2
	link: port:5 link:down
VLAN 1:
	vid: 1
	ports: 0t 4 
VLAN 2:
	vid: 2
	ports: 0t 5 
VLAN 20:
	vid: 20
	ports: 0t 4t 
VLAN 21:
	vid: 21
	ports: 0t 4t 
VLAN 22:
	vid: 22
	ports: 0t 4t 
VLAN 23:
	vid: 23
	ports: 0t 4t 
```

Reply:
Yes, however this statement resolves to "port 4 and port 5 are connected to eth0". This is what I'm referring to as not 
correct.

The way GMAC0 and GMAC1 work on ipq40xx is that they are abstracted using VID 1 and VID1 (WAN). Both are connected to port0 of the ESS-Switch. So it should be

```
"0u@eth0" "4:lan" "0u@eth1" "5:lan"
```

However, from my perspective changing the port bitmap for LAN and WAN on switch and GMAC level is the better solution. Otherwise, traffice between ports on the device has to pass the CPU, potentially impacting performance.

Have a look at https://git.openwrt.org/?p=openwrt/staging/blocktrron.git;a=commitdiff;h=1893ffdacce764638daf12d9e3960ded96bc8a57;hp=d6162698df6fd9f0d46c0186be9f5180c6e692b4 

It's not tested yet but from a performance standpoint, it's most likely the better solution. We can also ditch gmac1 with that.

Reply:
Ok, I understand. Probably I did not notice this, because I have always worked with VLANs on eth0. 
Since OpenWrt 19.07.0 I had no problems with the changes I made on the AVM 3000, unfortunately I have no more experience with OpenWrt.

Reply:
@blocktrron 

`"0u@eth0" "4:lan:1" "5:lan:1"`

Is having `:1` suffix twice intended?

Reply:
The output of swconfig only shows the perspective of the switch. gmac0 and gmac1 are both connected to port 0 of the switch and abstracted to the user via VLANs.

Reply:
Will the PR be accepted? Because the problem of an undetected switch on an AVM 3000 still exists.

Reply:
Besides the content, note that you have several formal issues, e.g.
- Missing prefix for commit title
- Room for improvement with the precise wording of the title itself
- Missing Signed-off-by
- The block after the device you change should be move so alphabetic sorting is maintained in the file.